tools/snap/build_remote.py

```diff
@@ -18,12 +18,14 @@ def _execute_build(target, archs, status, workspace):
         'snapcraft', 'remote-build', '--launchpad-accept-public-upload', '--recover', '--build-on', ','.join(archs)
     ], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, cwd=workspace)
 
-    line = process.stdout.readline()
-    while line:
-        _extract_state(target, line, status)
-        line = process.stdout.readline()
+    while True:
```

Comment:
nit: I got nerdsniped on this a bit so bear with me.

I feel like we can simplify things here. Can't the rest of this function just be something like:
```
for line in process.stdout:
    _extract_state(target, output, status)
process.wait()
return process.returncode
```
A lot of the unofficial examples online seem to use `readline` to read from stdout, but I see no reason this needs to be done.

I also think an empty line will never be empty before `EOF` is reached because it will always at least contain a newline character and we'll block on one being written to the stream.

https://docs.python.org/3/library/subprocess.html#subprocess.Popen.returncode describes when `returncode` is set. Using `wait` should be redundant with the `for `loop (maybe unless `snapcraft` does something like closing its stdout), but making sure the process has finished doesn't seem like a bad idea to me. We could also just have `return process.wait()` as the final line.

Reply:
Indeed, even simpler!